tests/apidoc_version_test.go

```diff
@@ -0,0 +1,31 @@
+package tests
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestVersionStructure(t *testing.T) {
+	deployment := Deploy(t, "test_version", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: Version responds 200 OK with valid structure
+	t.Run("Version responds 200 OK with valid structure", func(t *testing.T) {
+		res := unauthedClient.MustDo(t, "GET", []string{"_matrix", "client", "versions"}, json.RawMessage(`{}`))
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			JSON: []match.JSON{
+				match.JSONKeyPresent("versions"),
+				match.JSONArrayEach("versions", func(val gjson.Result) error {
+					return nil
```

Comment:
You can do more here. The original tests don't assert anything on the elements, but we can at least make sure they are strings by doing:
```go
match.JSONArrayEach("versions", func(val gjson.Result) error {
    if val.Type != gjson.String {
        return fmt.Errorf("value isn't string: %s", val.Raw)
    }
    return nil
}
```

This uses the `gjson` library which we use a lot in Complement - see the functions/fields you can use [here](https://pkg.go.dev/github.com/tidwall/gjson#Result).